modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -166,6 +169,38 @@ func isAllowedAddress(ctx sdk.Context, receiver string, allowedAddrs []string) b
 	return false
 }
 
+// validateForwarding performs the validation of forwarding info.
+func validateForwarding(forwarding Forwarding, allowedForwarding []AllowedForwarding) error {
+	if forwarding.Unwind {
+		return errorsmod.Wrap(ErrInvalidForwarding, "not allowed automatic unwind")
+	}
+
+	if !isAllowedForwarding(forwarding.Hops, allowedForwarding) {
+		return errorsmod.Wrap(ErrInvalidForwarding, "not allowed hops")
```

Comment:
should we `Wrapf` the unauthorized hops?

Reply:
https://github.com/cosmos/ibc-go/pull/6693/commits/7411f2d16e9cce64c4ecf09c201e4ed5b8aaab5d